app/controllers/api/v1/messages_controller.rb

```diff
@@ -35,11 +35,7 @@ class MessagesController < Api::V1::ApiController
       param :scope, String, desc: "The type of record associated to the messages (order/offer/item)"
       def index
         @messages = apply_scope(@messages, params[:scope]) if params[:scope].present?
-        @messages = @messages.where(id: params[:ids].split(",")) if params[:ids].present?
-        @messages = @messages.where(offer_id: params[:offer_id].split(",")) if params[:offer_id].present?
-        @messages = @messages.where(order_id: params[:order_id].split(",")) if params[:order_id].present?
-        @messages = @messages.where(item_id: params[:item_id].split(",")) if params[:item_id].present?
-        @messages = @messages.with_state_for_user(User.current_user, params[:state]) if params[:state].present?
+        apply_filters
```

Comment:
I would suggest writing the method as a pure function (generally that's always better than writing a method that modifies the state)

e.g

```
messages = apply_scope(@messages, params[:scope]) if params[:scope].present?
messages = apply_filters(messages, params)
paginate_and_render(messages)
```

It's a general practice we want to push forward with :)